source/develop/release-management/features/sla/mom-declarative-language.html.md

```diff
@@ -0,0 +1,276 @@
+      - io.write_bytes_per_s > policy.io_threshold
+      - net.throughput > policy.net_threshold
+```
+
+The condition `io_or_net_overutilized` is defined as the logical `or` of three conditions.
+The objects `io`, `net` and `policy` are VM property objects and
+contain the properties the rules have access to.
+Other property objects can be `cpu`, `memory` or anything else.
+
+The next section defines rules.
+
+```yaml
+rules:
+    cpu_limit:
+        output: cpu.max_load
+        min: 10
+        target: 0
+        function:
+            name: linear
+            change: 1
+            time: 1 sec
+
+        when: io_or_net_overutilized
+```
+
+This rule is triggered when the condition `io_or_net_overutilized` is satisfied.
+The property `cpu.max_load` of the VM is linearly decreased by 1 percent per second.
+Until the minimum value `10` is reached.
+
+Meaning of the rule attributes:
+
+- `output` - The property that is modified by the rule.
+- `min` - The minimum allowed value of the output property. It can be any expression, not only a constant.
+- `target` - The ideal value we are trying to reach (can be any expression).
+- `function` - Defines how the `output` is changed in response to computed target.
+    - `name` - Name of the function
+    - `change`, `time` - Parameters of the function.
+- `when` - Condition, that has to be satisfied. It can be any boolean expression and use the predefined conditions.
+
+The second rule is:
+
+```yaml
+cpu_release:
+    output: cpu.max_load
+    max: 100
+    target: 100
+    function:
+        name: exponential
+        factor: 2
+        time: 30 sec
+
+    when: not io_or_net_overutilized
+```
+
+It is triggered when `io_or_net_overutilized` is not satisfied.
+The maximum CPU load of the VM is exponentially increasing.
+It doubles every 30 seconds, until the maximum 100 is reached.
+
+## More features of the representation
+
+### Vars block
+
+The block contains values used in the rules. An example from the KSM rule:
+
+```yaml
+vars:
+    ksm_pages_boost: 300
+    ksm_pages_decay: -50
+    ksm_npages_min: 64
+    ksm_npages_max: 1250
+    ksm_sleep_ms_baseline: 10
+    ksm_free_percent: 0.20
+```
+
+An example from the ballooning rule:
+
+```yaml
+vars:
+    host_free_percent: host.mem_free_avg / host.mem_available
+    vm_used_mem: memory.balloon_cur - memory.unused
+```
+
+### Minimum needed change and maximum allowed change
+
+In some cases it may not be practical to modify a value if the change would be too small.
+
+- `min_absolute_change` - Will not modify the output if the absolute change is smaller.
+- `max_absolute_change` - Limits the absolute change.
+- `min_relative_change` - Will not modify the output if the relative change is smaller.
+- `max_relative_change` - Limits the relative change.
+
+```yaml
+rules:
+  balloon_shrink:
+    output: memory.balloon
+    min: policy.balloon_min
+    min_relative_change:
+        value: min_balloon_change_percent
+        interval: 10 sec
```

Comment:
The minimal change does not need the `interval` property. It just won't be saved until the current difference is big enough.

The maximal allowed change needs the interval as it is part of the limit.

Reply:
Fixed.